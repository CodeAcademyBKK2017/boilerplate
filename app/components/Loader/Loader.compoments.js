import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Loader.compoments.style';
import {ActivityIndicator, Modal, View
} from 'react-native';

export default class Loader extends Component {
  render () {
    return (
      <Modal transparent={true} visible={this.props.show} style={[styles.container, styles.horizontal]}>
        <View  style={styles.containerView}>
          <ActivityIndicator animating={this.props.show} size='large' color='#0000ff' />
        </View>
      </Modal>
    );
  }
}

Loader.propTypes = {
  show: PropTypes.bool
};
  