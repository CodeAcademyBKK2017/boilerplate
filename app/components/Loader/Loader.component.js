import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Loader.style';
import {ActivityIndicator, Modal, View} from 'react-native';

export default class Loader extends Component {
  render () {
    return (
      <Modal
        visible={this.props.visible}
        animationType={'fade'}
        transparent={true}>
        <View style={styles.container}>
          <ActivityIndicator size='large'/>
        </View>
      </Modal>
    );
  }
}

Loader.propTypes = {
  visible: PropTypes.bool.isRequired
};

Loader.defaultProps = {
  visible: false
};
