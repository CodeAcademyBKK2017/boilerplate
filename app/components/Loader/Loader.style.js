import {
  StyleSheet
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#00000033'
  },
  horizontal: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10
  }
});

export default styles;
  