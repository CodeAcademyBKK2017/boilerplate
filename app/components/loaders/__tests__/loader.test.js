import Loader from '../loader.component';
import React from 'react';
import renderer from 'react-test-renderer';
import  'react-native';

// Note: test renderer must be required after react-native.
// import uuid from 'uuid';;

describe('Loader', () => {

  it('Loader: renders correctly', () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});