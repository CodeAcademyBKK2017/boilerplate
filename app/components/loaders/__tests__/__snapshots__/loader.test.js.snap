// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Loader: renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "background: rgba(170, 169, 169, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#ff0000"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</Modal>
`;
